driver/Configuration.ml

```diff
@@ -82,9 +94,19 @@ let get_config_list key =
   | [] -> bad_config key []
   | vl -> vl
 
-let prepro = get_config_list "prepro"
-let asm = get_config_list "asm"
-let linker = get_config_list "linker"
+let tool_absolute_path tools =
+  match tools with
+  | [] -> []
+  | tool::args -> let tool =
+      if is_implicit tool && dirname tool = current_dir_name then
```

Comment:
`Filename.is_implict "a/b"` returns true as well, but `Filename.dirname "a/b"` is equal to `a`.